### ISSUE TITLE
hx-get with mixed <script src=""> and <script>src</script>

### ISSUE THREAD

**Issue by BoPeng** [bigskysoftware/htmx#787]
Continuation of #379

`hx-get` content with `<script src="">` and `<script>src</script>` appear to work now, but

```html
<script src="url_of_package"></script>
<script>
function_from_package()
</script>
```
does not appear to work. It looks like the function is called before `url_of_package` is loaded.

**Comment by BoPeng** [comment]
I did a lot of testing and reading, and posts like [this](https://stackoverflow.com/questions/8996852/load-and-execute-order-of-scripts) do not really help. I, however, notice that the inline script will be executed after the `<script src=>` is loaded if `type="module"` is added as follows.

```html
<script src="url_of_package"></script>
<script type="module">
function_from_package()
</script>
```

This requires, however, that 

https://github.com/bigskysoftware/htmx/blob/f1cef92e528871993df1a5569b30bb94c99ec847/src/htmx.js#L1434

is changed to

```javascript
            if (script.type === "text/javascript" || script.type === "" || script.type=== 'module') {
```

I do not know why `module` was left out but this appears to be an innocent change. @1cg , would you consider merging a PR for this?

**Comment by BoPeng** [comment]
Thanks for merging the PR.

**Comment by morindo** [comment]
Greetings @BoPeng, did you find a permanent solution for this issue? 

I'm currently migrating a legacy app to HTMX, and it's just awesome. I'm also using the **head-support** extension to merge new stuff (like scripts and css) in the head section. Currently my pages body inline scripts (without src attribute) execute before newly added head scripts from a **hx-boost** on page load. Do you know if there is a reason for that? Am I doing something wrong?

```
<a href="/mypage.html" hx-boost="true">My Page</a>

<html>
<head>
<title>My Page</title>
<!-- script1.js contains a function called showMe() -->
<script src="script1.js" />
</head>
<body hx-ext="head-support">
<script>
$(function() {
showMe(); // This function get called before script1.js is evaluated by the browser.
});
<script>
</body>
</html>
```

My current htmx version is 1.9.8.

**Comment by BoPeng** [comment]
No, I have not solved my problem completely. My solution is currently making sure that `<script src="script1.js" />` is executed before the `<script>script</script>` is called. Basically, assuming I have a page with a `hx-get` link, I am doing the following to make sure `scritp1.js` is executed before the link is clicked (and its `<script>my script</script>` is called:

```html
<a hx-get="/another/url/with_script">click me </a>

<script src="script1.js" />
```

**Comment by morindo** [comment]
Ok, thank you very much, I'll look into it.

**Comment by morindo** [comment]
@BoPeng finally, there are no problem with HTMX regarding the order of execution of scripts. It's mostly a browser problem/feature async code execution, really well explain here.

Deep dive into the murky waters of script loading
https://web.dev/articles/speed-script-loading

Thanks again!

**Comment by morindo** [comment]
Sorry for the late response @BoPeng. In your case, I think the only thing you can do is put the calling_dependency() function into an external file. My recommendation is to use external scripts only or, not recommended, but should work, use successive inline scripts.

```
<div hx-get="/something" hx-target="#id1"></div>

<div id="id1">
    <span>...</span>
    <script src="dependency.js"></script>
    <script src="calling_dependency.js"></script>
</div>

or

<div hx-get="/something" hx-target="#id1"></div>

<div id="id1">
    <span>...</span>
    <script>function myfunction() { alert('This should work!'); }</script>
    <script>myfunction();</script>
</div>
```

Also, you could check for **htmx.config.useTemplateFragments** option. I had a case when htmx.config.useTemplateFragments that was set to **true**, inline script function were called immediately after the insertion into the DOM. In fact, my inline script was executed twice once at the insertion into the DOM and the other from the htmx processScripts/evalScript function. Setting the htmx.config.useTemplateFragments to **false** seems to defer the script execution and execute only once.
